app/presenters/disconnect_presenter.rb

```diff
@@ -0,0 +1,26 @@
+class DisconnectPresenter < WarningPresenter
+  attr_reader :destination, :messages
+
+  def initialize(destination)
+    @destination = destination
+    @messages = {
+      both_pages: I18n.t('pages.disconnecting_modal.both_pages'),
+      cya_page: I18n.t('pages.disconnecting_modal.cya'),
+      confirmation_page: I18n.t('pages.disconnecting_modal.confirmation')
+    }
+    super(updated_service)
+  end
+
+  private
+
+  # we need a service which represents what the metadata WOULD look like
+  # if the user changed the destination
+  def updated_service
+    @updated_service ||=
+      MetadataPresenter::Service.new(destination.metadata)
+  end
+
+  def grid
+    @grid ||= MetadataPresenter::Grid.new(updated_service)
+  end
```

Comment:
in the `initialize` i think you can do 

```
@service = updated_service
super(service)
```

this this overriding grid method can be removed